src/pages/index.js

```diff
@@ -31,44 +36,96 @@ const cardImagePopup = new PopupWithImage("#preview-image-modal");
 cardImagePopup.setEventListeners();
 
 
+
 /*                                                                          */
 /*                             FUNCTIONS                                    */
 
 function handleCardFormSubmit(inputValues) {
   const name = inputValues.title;
   const link = inputValues.link;
-  renderCard({ name, link });
-  addCardPopup.close();
-  addCardForm.reset();
-  addCardFormValidator.toggleButtonState();
+  // renderCard({ name, link });
+  addCardPopup.renderLoading(true);
+  api.createNewCard(name, link)
+    .then((data) => {
+      renderCard(data);
+      addCardPopup.close();
+      addCardForm.reset();
+      addCardFormValidator.toggleButtonState();
+    })
+    .catch((err) => {
+      console.log(err);
+    }).finally(() => {
+      addCardPopup.renderLoading(false);
+
+    })
+}
+const handleCardLike = (cardId, cardInstance) => {
+  const isItLiked = cardInstance.like;
+  console.log(cardInstance.like);
+  const toggleLike = isItLiked ? api.likeButtonDeactive(cardId) : api.likeButtonActive(cardId);
+  toggleLike
+    .then((updatedCardData) => {
+      cardInstance.handleLike(!isItLiked);
+    })
+    .catch((err) => {
+      console.error("like toogle error:", err)
+    })
 }
-
 function createCard(cardData) {
   const cardElements = new Card(cardData, "#card-template", () => {
-    cardImagePopup.open(cardData);
-  });
+    cardImagePopup.open(cardData)
+  },
+    (cardId) => {
+      deleteCardPopup.open(cardId);
+
+    }, handleCardLike,
+  );
+
   return cardElements.viewCard();
-}
+};
 
 function renderCard(cardData) {
   const newCard = createCard(cardData);
+  newCard.id = cardData._id;
   cardSection.addItem(newCard);
 }
 
 
 /*                                                                          */
 /*                                EVENT LISTENERS                           */
 /*                                                                          */
+/*change profile picture*/
+
 
+editProfileImageIcon.addEventListener("click", () => {
+  changeProfilePhoto.open();
+})
 /*-------------------------------PROFILE ----------------------------------*/
 profileEditButton.addEventListener("click", () => {
   const { title, job } = profileInfo.getUserInfo();
   profileTitleInput.value = title;
   profileDescriptionInput.value = job;
   profileEditPopup.open();
 })
-
-
+const changeProfilePhoto = new PopupWithForm({
+  popupSelector: "#change__profile_picture",
+  handleFormSubmit: (data, submitButton) => {
+    changeProfilePhoto.renderLoading(true);
+    api.profilePicture(data.avatar)
+      .then((res) => {
+        profileInfo.setAvatar(res.avatar);
+        submitButton.classList.add("modal__button_disabled");
```

Comment:
`        submitButton.classList.add("modal__button_disabled");
`

Now only `FormValidator` should **`disable/enable`** submit buttons and remove errors from the inputs (if you do it). It’s a violation of **Single responsibility** principle


```js

      .then((res) => {
        profileInfo.setAvatar(res.avatar);
        avatarForm.reset()  // reset the form first
        changeAvatarFormValidator.toggleButtonState() // this will disable the avatar save button after resetting
        changeProfilePhoto.close();
      })
```